### PR TITLE
WIP: Hub Guides

### PR DESCRIPTION
**Pull request by jared-s**
PR for collaborative Guide writing

### CODE REVIEW

**Comment by jared-s** [comment]
@mrinalwadhwa for these new guides @mjc and I are working on..did you want them in a new Hub examples How To Guide section, or still under Rust SDK? For now I split it out, but easy to combine

**Comment by mrinalwadhwa** [comment]
Still in rust ask

**Comment by jared-s** [comment]
Removed `echo.md` because i noticed that the nodes and workers example is already the echo example, that we can edit

**Comment by mjc** [comment]
We should probably ask the user to get their test node connection information from https://hub.ockam.network instead of providing an example IP (nodes-and-hub.md: 62, 91.)

Curious also if y'all think we should have them use `.to_socket_addrs()` (or whatever the current way to do that in Rust is) to get an IP from the hostname of their node?  In my elixir snippets (that I plan on turning into code examples) I’m doing something similar with `:inet.gethostbyname/1` so they don’t have to hardcode IPs.

**Comment by jared-s** [comment]
> We should probably ask the user to get their test node connection information from https://hub.ockam.network instead of providing an example IP (nodes-and-hub.md: 62, 91.)

Agreed, will do this.

> 
> Curious also if y'all think we should have them use `.to_socket_addrs()` (or whatever the current way to do that in Rust is) to get an IP from the hostname of their node? In my elixir snippets (that I plan on turning into code examples) I’m doing something similar with `:inet.gethostbyname/1` so they don’t have to hardcode IPs.

Will search for the current most pragmatic way, we have been avoiding hostnames in favor of ip so we can be lazy and use SockAddr::parse :P

**Comment by mjc** [comment]
> > Curious also if y'all think we should have them use `.to_socket_addrs()` (or whatever the current way to do that in Rust is) to get an IP from the hostname of their node? In my elixir snippets (that I plan on turning into code examples) I’m doing something similar with `:inet.gethostbyname/1` so they don’t have to hardcode IPs.
> 
> Will search for the current most pragmatic way, we have been avoiding hostnames in favor of ip so we can be lazy and use SockAddr::parse :P

💯 and I'm only using it in my elixir demonstration code so that I won't forget what the IP is actually for.  It only occurred to me while setting up the UI that we probably want to be able to change IPs for services in the future.

Looking forward to what you come up with, I definitely still struggle with finding the "right" way to do things in Rust and DNS seems to have several different ways to do it that I've seen.

**Comment by jared-s** [comment]
@mjc can I get a screenshot of what the user sees when logging into hub to retrieve their connectivity information? When I try to hit hub I'm getting `Failed to authenticate` so guessing it's not quite ready?

**Comment by jared-s** [comment]
I think we should provide a little utility function like so:

```rust
fn get_hub_address(host_and_port: &str) -> SocketAddr {
    if let Ok(addrs) = host_and_port.to_socket_addrs() {
        if addrs.len() != 0 {
            return addrs.as_slice()[0];
        }
    }
    panic!("Unable to resolve Ockam Hub address :(");
}
```

Alternatives require bringing in crates, and doesn't seem any of the popular resolver crates are transitive dependencies yet.

**Comment by mjc** [comment]
> @mjc can I get a screenshot of what the user sees when logging into hub to retrieve their connectivity information? When I try to hit hub I'm getting `Failed to authenticate` so guessing it's not quite ready?

it worked until we enabled SSL. let me see if I can get that changed...  I also am working on updating it to read slightly more professionally now that we have the guide.

File: src/content/learn/how-to-guides/rust-sdk-code-examples/nodes-and-hub.md [hunk 1]
```diff
@@ -7,23 +7,56 @@ order: 3
...
+## Getting started with Ockam Hub
+
+Ockam Hub is a service provided by Ockam that allows you to easily test Ockam Node based networks.
+
+Follow these steps to get started with Ockam Hub:
+
+1. Sign in to Ockam Hub at (https://hub.ockam.network/)[https://hub.ockam.network/] with your GitHub account.
+1. After your Ockam Node is deployed, you will see a confirmation page with instructions.
+   1. Find and save the hostname of the node. You will use this hostname in your code to build a route to the Hub.
+   1. Alternatively, you can download a pre-configured code snippet by clicking the '(??)' button.
```
Comment (jared-s) [inline src/content/learn/how-to-guides/rust-sdk-code-examples/nodes-and-hub.md:19]: @mjc had the idea of providing a button which lets the user download a pre-configured snippet, so they don't have to worry about copying and pasting the address. I like the idea. Just need to know the final button name here

Comment (mjc) [reply src/content/learn/how-to-guides/rust-sdk-code-examples/nodes-and-hub.md:19]: I was thinking about it just displaying the modified snippet, but we could actually do both. Downloading sounds like a useful thing.

Comment (jared-s) [reply src/content/learn/how-to-guides/rust-sdk-code-examples/nodes-and-hub.md:19]: Cool, also by "download" I was thinking also a "copy to clipboard" button might be good, and maybe faster to implement than a full download with filenames and the like

Comment (mjc) [reply src/content/learn/how-to-guides/rust-sdk-code-examples/nodes-and-hub.md:19]: Currently the text is "Get a preconfigured echo service example" and "Get a preconfigured forwarding service example"

**Comment by jared-s** [comment]
Merging so i can screenshot, to be continued in future PR

**Pull request merged**